services/requester/single_tx_pool.go

```diff
@@ -193,37 +209,37 @@ func (t *SingleTxPool) buildTransaction(
 	}
 
 	// now that the transaction is prepared, store the transaction's metadata
-	accKey.SetLockMetadata(flowTx.ID(), latestBlock.Height)
+	accKey.SetLockMetadata(flowTx.ID(), latestBlockHeader.Height)
 
 	t.collector.OperatorBalance(account)
 
 	return flowTx, nil
 }
 
-func (t *SingleTxPool) fetchFlowLatestBlockAndCOA(ctx context.Context) (
-	*flow.Block,
-	*flow.Account,
-	error,
-) {
-	var (
-		g           = errgroup.Group{}
-		err1, err2  error
-		latestBlock *flow.Block
-		account     *flow.Account
-	)
+func (t *SingleTxPool) trackLatestBlockHeader(ctx context.Context) {
+	ticker := time.NewTicker(blockHeaderUpdateFrequency)
+	defer ticker.Stop()
 
-	// execute concurrently so we can speed up all the information we need for tx
-	g.Go(func() error {
-		latestBlock, err1 = t.client.GetLatestBlock(ctx, true)
-		return err1
-	})
-	g.Go(func() error {
-		account, err2 = t.client.GetAccount(ctx, t.config.COAAddress)
-		return err2
-	})
-	if err := g.Wait(); err != nil {
-		return nil, nil, err
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			blockHeader, err := t.client.GetLatestBlockHeader(ctx, false)
+			if err != nil {
+				t.logger.Error().Err(err).Msg(
+					"failed to update latest Flow block header",
+				)
+				continue
+			}
+			t.latestBlockHeader.Store(blockHeader)
+		}
 	}
+}
 
-	return latestBlock, account, nil
+func (t *SingleTxPool) getLatestBlockHeader() *flow.BlockHeader {
```

Comment:
I'd suggest framing this as `getReferenceBlockID()`. that way future devs don't expect it to return the actual latest block, which this doesn't.

Reply:
Good suggestion, updated in https://github.com/onflow/flow-evm-gateway/pull/896/commits/3f378f3132cb9c4da87c47be5edc5eaf178bd4c7 , along with all references which implied that we are dealing with the latest block instead, when this is updated every 15 seconds, and only supposed to be used as the reference block ID.